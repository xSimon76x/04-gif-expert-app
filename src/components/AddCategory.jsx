
export const AddCategory = ({onAddAnime, inputValue, setInputValue}) => {

    const agregarAnime = (e) => {
        e.preventDefault();
        if(inputValue.trim().length <= 1) return;

        onAddAnime(inputValue.trim())
        setInputValue("");
    }
    

    return (
        <>  
            <form onSubmit={(e) => agregarAnime(e)}>
                <div style={{display: "flex"}}>
                    <input 
                        id="inputCategoria" 
                        type="text"
                        onChange={ e => setInputValue(e.target.value)}
                        value={inputValue}
                    />
                    <button onClick={agregarAnime}>Agregar Anime</button>
                </div>
            </form>

        </>
    )
}