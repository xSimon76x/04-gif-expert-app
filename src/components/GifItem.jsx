import React from 'react';
import ReactDOM from 'react-dom/client';
import PropTypes from 'prop-types';

export const GifItem = ({title, url}) => {
  return (
    <div className="card">
      <p>{title}</p>
      <img src={url} alt={title} />
    </div>
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
  //url: PropTypes.string.isRequired
}