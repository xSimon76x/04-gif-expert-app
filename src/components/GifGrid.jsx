
import { useFetchGift } from "../hooks/useFetchGift";
import { GifItem } from "./GifItem";

export const GifGrid = ({inputValue}) => {
  
    const { images, isLoading } = useFetchGift(inputValue);

    console.log(inputValue)

    return (
        <>
            <div className="card-grid">
                {
                    !isLoading ? Object.values(images).map( (image) => (
                        <GifItem key={image.id} {...image} />
                    ))
                    :
                    <p>cargando...</p>
                }
            </div>
            
        </>
    )
}
