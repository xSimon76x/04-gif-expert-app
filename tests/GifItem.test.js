import ReactDOM from 'react-dom/client';
import { render, screen } from '@testing-library/react';
import { GifItem } from "../src/components/GifItem";


describe('GifItem - Test', () => {

    const title = "Goku";
    const url = "https://www.latercera.com/resizer/kMBcLnUVzd4zyy1YYTmqtjhJK4o=/768x0/smart/filters:quality(70):format(webp):no_upscale()/arc-anglerfish-arc2-prod-copesa.s3.amazonaws.com/public/DTJTGBAZKBBNBD5Q3WFW25YXU4.gif";
    
    test('debe obligar a requerir los campos de url y titulo', () => {
        
        
        const { container, getByText, getByTestId } = render(
            <GifItem title={title} url={url} />
            )
            
            expect( container ).toMatchSnapshot();
            
            
        });
        
    test('debe mostrar la imagen con el URL y el ALT indicado', () => {

        render( <GifItem title={title} url={url} /> );

        // screen.debug();

        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);

        const { src, alt } = screen.getByRole('img');

        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('debe existir el titulo en el componente', () => {
        
        render(
            <GifItem title={title} url={url} />
        );

        expect(screen.getByText(title)).toBeTruthy();
    });
})
