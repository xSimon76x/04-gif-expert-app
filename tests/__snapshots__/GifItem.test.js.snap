// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem - Test debe obligar a requerir los campos de url y titulo 1`] = `
<div>
  <div
    class="card"
  >
    <p>
      Goku
    </p>
    <img
      alt="Goku"
      src="https://www.latercera.com/resizer/kMBcLnUVzd4zyy1YYTmqtjhJK4o=/768x0/smart/filters:quality(70):format(webp):no_upscale()/arc-anglerfish-arc2-prod-copesa.s3.amazonaws.com/public/DTJTGBAZKBBNBD5Q3WFW25YXU4.gif"
    />
  </div>
</div>
`;
